File: server/app/models/audit.py
from typing import Optional, Union
from beanie import Document


class AuditLog(Document):
    aid: str
    userId: Optional[int] = None
    userName: Optional[str] = None
    action: str
    entity: Optional[str] = None
    entityId: Optional[Union[str, int]] = None
    details: Optional[str] = None
    timestamp: str
    ipAddress: Optional[str] = None

    class Settings:
        name = "audit_logs"

File: src/types/audit.ts
// src/types/audit.ts

export type AuditAction =
  | 'LOGIN'
  | 'LOGOUT'
  | 'CREATE'
  | 'UPDATE'
  | 'DELETE'
  | 'VIEW'
  | 'SCHEDULE'
  | 'UPLOAD';

export interface AuditLog {
  id: string | number;
  userId: number;
  userName: string;
  action: AuditAction;
  entity: string;
  entityId?: string | number;
  details?: string;
  timestamp: string;
  ipAddress?: string;
}

export interface AuditState {
  logs: AuditLog[];
  isLoading: boolean;
  error: string | null;
}
